DocJuris

joshuaapple opened the pull request:
- Add types and endpoints for v2 of API
- Deprecate certain v1 endpoints
- Clean up some incorrect descriptions
Update schema definition for DocJuris connector

---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [ x ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ x ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ x ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ x ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ x ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ x ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

joshuaapple committed:
Update schema definition for DocJuris connector

- Add types and endpoints for v2 of API
- Deprecate certain v1 endpoints
- Clean up some incorrect descriptions

vmanoharas commented:
Hello @joshuaapple, 

Could you please confirm that the existing connections will not be broken due the breaking changes shown in the swagger validation errors?

Additionally, it seems that this connector is Verified publisher connector, please remove the Independent Publisher from the PR title 

##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/TurnCompletedTrigger/responses/x-ms-notification-content/schema/turnId/format Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/TurnCompletedTrigger/responses/x-ms-notification-content/schema/contractId/format Source: NewSwagger
##[error]The operation has been deleted. This change breaks existing flows. ValuePath: operation/TurnTaggedTrigger Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/TurnCreatedTrigger/responses/x-ms-notification-content/schema/turnId/format Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/TurnCreatedTrigger/responses/x-ms-notification-content/schema/contractId/format Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/ScreeningReportUpdatedTrigger/responses/x-ms-notification-content/schema/turnId/format Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/ScreeningReportUpdatedTrigger/responses/x-ms-notification-content/schema/contractId/format Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/TurnMentionTrigger/responses/x-ms-notification-content/schema/turnId/format Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/TurnMentionTrigger/responses/x-ms-notification-content/schema/contractId/format Source: NewSwagger

joshuaapple commented:
- Removed the (Independent Publisher) annotation
- The list changes resulted from a "uuid" format annotation added to 2 types. It did not affect anything in our testing, but since there maybe ramifications that I'm not aware of, and they aren't necessary for anything, I've removed them.

vmanoharas commented:
Hi @joshuaapple, 

Thank you for your response!

We are checking again that this connector is already in production, so the changes in the operation IDs would affect for the existing customers, not in your testing.  Please confirm, before we proceed futher.

joshuaapple commented:
Maybe I've misunderstood- are you asking about the swagger validation errors you listed? I removed the change that caused those errors, so it should no longer be an issue. Am I missing something else?

vmanoharas commented:
Hi @joshuaapple, 

Yes we are asking swagger validation errors listed below (latest updates), please confirm that these errors will not break the existing connections. 

##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/TurnCompletedTrigger/responses/x-ms-notification-content/schema/turnId/format Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/TurnCompletedTrigger/responses/x-ms-notification-content/schema/contractId/format Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/TurnCreatedTrigger/responses/x-ms-notification-content/schema/turnId/format Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/TurnCreatedTrigger/responses/x-ms-notification-content/schema/contractId/format Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/ScreeningReportUpdatedTrigger/responses/x-ms-notification-content/schema/turnId/format Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/ScreeningReportUpdatedTrigger/responses/x-ms-notification-content/schema/contractId/format Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/TurnMentionTrigger/responses/x-ms-notification-content/schema/turnId/format Source: NewSwagger
##[error]The type format has been changed for the output field. Existing flows may be broken if they are using this field and actual payload has been changed. ValuePath: operation/TurnMentionTrigger/responses/x-ms-notification-content/schema/contractId/format Source: NewSwagger

joshuaapple commented:
Like I said, I removed the field annotations that caused those warning, so no, it will not affect existing connections.

vmanoharas reviewed (approved):
Dear Partner, 

Congratulations your PR is approved !! 

As next steps: 

1.	Learn about the new certification experience- [Get your Power Platform connector and plugin certified - Overview](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fsubmit-certification&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390456966%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=uWDvs%2BmTbozw1oPc%2FNDNWJ%2FHgnUSfFlRCB1T8UHBh3A%3D&reserved=0)
2.	Package your connector and plugin files- [Prepare Power Platform connector and plugin files for certification](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-submission&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390465418%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=pDuv4bnya1eP3Rrs3p4ArNOXw%2FQipWU87q5oZ0VSIe0%3D&reserved=0)
3.	Validate the package for structure before submitting the package for certification to Partner Center - [Run Package Validator tool](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-submission%23step-7-validate-the-package-for-structure&data=05%7C02%7Cshefaali.patankar%40microsoft.com%7C6f0c4adf688f42a7107908dcb6c72a8c%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638586213737574171%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=9QGRIRe%2FZ%2B1sgUF5JQjY2DeuyxDt8f8lDwpGFEaaBmg%3D&reserved=0) 
4.	Initiate a certification request in Partner Center- [Verified publisher certification process](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fsubmit-for-certification&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390471799%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=zbm2px5VZ2ABLBFcORH6tu1fsC9KWdXkP1XJFyluMzw%3D&reserved=0)
5.	Ensure your connector & plugin files comply to the Marketplace policies here- [5000 Power Platform Connector Policies for Marketplace](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Flegal%2Fmarketplace%2Fcertification-policies%235000-power-platform-connectors&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390477736%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=h0pI5J5buo82xxvnBVmrAN8K%2F%2BK8xZlUdcSwzWoqWq0%3D&reserved=0) and also [1000 Marketplace policies](https://learn.microsoft.com/en-us/legal/marketplace/certification-policies#100-general)(as applicable) 
6.	Test your connector post certification to provide go-live for deployment- [Testing Guidelines](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-testing&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390483743%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=Z8pSnhCArFRzEr373o29qM5h5VbV1nlN2gVbv%2Blk1aE%3D&reserved=0)
7.	If you wish to Update your connector or plugin in future- [Updating Guidelines](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-updates&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390489681%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=CvxTWJFE3bTFeJ3nDJoCSpzvimhVV9go4cBG%2FQ4gMvo%3D&reserved=0)

For any queries or concerns please get in touch with us on [certificationteam@microsoft.com](mailto:certificationteam@microsoft.com)

Many thanks,
Power Platform 
Copilot connector & plugin certification team
